.github/workflows/back-compat-pr.yml

```diff
@@ -0,0 +1,43 @@
+name: back-compat PR
+
+on:
+  pull_request:
+    paths:
+      - "go.*"
+      - "**/*.go"
+      - ".github/workflows/*.yml"
+
+permissions:
+  contents: read
+
+env:
+  GO_VERSION: '1.22.5'
+
+jobs:
+  check-back-compat:
+    name: build
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+    - name: Set up Go
+      uses: actions/setup-go@v5
+      with:
+        go-version: ${{ env.GO_VERSION }}
+    - name: build head
+      run: |
+        go build -o omm_head
+        cp omm_head /var/tmp
+    - uses: actions/checkout@v4
+      with:
+        ref: main
+    - name: build main
+      run: |
+        go build -o omm_main
+        cp omm_main /var/tmp
+    - name: Run last version
+      run: |
+        /var/tmp/omm_main --db-path=omm-test.db 'test: a task from main'
```

Comment:
Cannot this be covered by a test?

Could what I [suggested](https://github.com/dhth/omm/pull/23#discussion_r1702916008) to fix some issues that led you doing this here?

Reply:
This is for ensuring any changes to the data persistence side of things (database migrations, handling of errors) do not mess anything up for already running instances of omm. Specifically, for ensuring that a change doesn't break for folks who already have a `omm.db` file out there.

I guess this may be tested via code tests (would need simulating real world conditions in tests); running the real thing in CI seemed like a good first step.

Reply:
Eg. for cases caught by [this](https://github.com/dhth/omm/actions/runs/10231339314/job/28307166074?pr=23) run.